lib/cfilters.c

```diff
@@ -168,6 +177,61 @@ void Curl_conn_close(struct Curl_easy *data, int index)
   }
 }
 
+CURLcode Curl_conn_shutdown_blocking(struct Curl_easy *data, int sockindex)
```

Comment:
This risks blocking the multi interface for the entire timeout period (DEFAULT_SHUTDOWN_TIMEOUT_MS) when the FTPS data connection is closed. Do you plan to add handling of this in a non-blocking way in a follow-up change?

Reply:
To have a non-blocking shutdown of the data connection would be nice, indeed. That would require some more radical change in the FTP code, I guess.

For now, the blocking way at least assures that the FTP data connection works. We have multiple reports that at least uploads fail regularly if we do not do this, as FTP servers have become more sensitive to a clean TLS shutdown.

The change in this PR is a bugfix at the expense of parallel performance.

Reply:
Yeah. Let's merge this as it fixes issues, and maybe work on improving it further later on.